icicle_v3/backend/cpu/include/hash/blake2/blake2s.h

```diff
@@ -0,0 +1,117 @@
+#ifndef BLAKE2_H
+#define BLAKE2_H
+
+typedef unsigned char BYTE;
```

Comment:
I think you should put these typedefs into a namespace or remove them, as this can clash with other names in a global namespace